Guide to Setting Up TextLocal as Your Phone Login Provider (#21499)
supabase/supabase: apps/docs/content/guides/auth/phone-login/textlocal.mdx

---
id: 'auth-textlocal'
title: 'Phone Auth with Textlocal'

> description: 'How to set up and use Mobile OTP with Textlocal and Supabase.'

Reviewer: ```suggestion
subtitle: 'How to set up and use Mobile OTP with Textlocal and Supabase.'
description: 'How to set up and use Mobile OTP with Textlocal and Supabase.'
```

---

<Admonition type="tip">

> To manage costs related to SMS authentication, consider setting project rate limits and configuring CAPTCHA. For additional cost-control strategies and to ensure compliance with TRAI's DLT requirements, visit the Production Checklist.

Reviewer: Pretty sure we don't have anything about TRAI DLT in the Production Checklist.

Also, is this an India-specific requirement? If so, maybe we could make that more clear, to avoid confusing users from other countries and make it more globally applicable.

```
Also check country-specific regulations around sending SMS messages, for example, India's TRAI DLT regulations.
```

</Admonition>

## Prerequisites

Before you begin, ensure you have the following:

> - **A Textlocal account**: Sign up [here](https://www.textlocal.com/signup) to start sending SMS.

Reviewer: ```suggestion
- **A Textlocal account**: Sign up for [TextLocal](https://www.textlocal.com/signup) to start sending SMS messages.
```

> - **A Supabase project**: Necessary for SMS authentication integration. Create your project [here](https://supabase.com/dashboard).

Reviewer: ```suggestion
- **A Supabase project**: Necessary for SMS authentication integration. Create your project in the [Supabase Dashboard](https://supabase.com/dashboard).
```

> - **A mobile phone**: To receive SMS, verifying the setup and functionality.

Reviewer: ```suggestion
- **A mobile phone**: To receive SMS messages and test your setup.
```

> - **DLT Registration**: Required for businesses in India, as per TRAI regulations, to send commercial SMS.

Reviewer: Can you take the DLT registration instructions out in favor of the admonition above? ☝🏼 

We don't have the capacity to keep up-to-date instructions on how to comply with government regulations, and don't want to provide potentially outdated ones, especially given the number of countries we serve.

## DLT Registration Requirements for India

The Telecom Regulatory Authority of India (TRAI) mandates that all businesses intending to send commercial SMS in India must register their business entity, headers (sender names), and message templates on DLT platforms. This step is essential for legally and effectively sending SMS within India, ensuring compliance with TRAI regulations.

1. **Complete your DLT registration**: Visit [this link](https://www.textlocal.in/tcccpr-trai-new-dlt-registration/) for a detailed guide on registering your business on a DLT platform.
2. **Update your Textlocal account**: After receiving your Entity ID/Registration ID from the DLT registration, update it in your Textlocal account settings [here](https://control.textlocal.in/settings/account/).
3. **Approval confirmation**: Share a screenshot showing your approved headers from the DLT platform with Textlocal support at support@textlocal.in to activate your SMS services.

> ## Setting Up Textlocal as Your SMS Provider

Reviewer: ```suggestion
## Setting up Textlocal as your SMS provider
```

> To integrate Textlocal with Supabase, you'll need:
> 
> - **API Key**: For API authentication.
> - **Sender Name**: The name displayed as the SMS sender.
> - **Supabase Configuration**: Steps to integrate with your Supabase project.

Reviewer: ```suggestion
To integrate Textlocal with Supabase:

1. [Get a Textlocal API key](#get-a-textlocal-api-key)
1. [Customize your sender name](#customize-your-sender-name-optional)
1. [Configure your Supabase project](#configure-supabase)
```

> ### Acquire Your Textlocal API Key

Reviewer: ```suggestion
### Get a Textlocal API key
```

> First, log into your Textlocal account and head to the API Keys section under Settings. Here, generate a new API Key. This key will facilitate secure communication between your application and Textlocal's SMS gateway.

Reviewer: ```suggestion
1. Log into your Textlocal account and go to `Settings` > `API Keys`.
1. Generate a new API Key. Save your new API key in a safe location.
```

> ![Get your API Keys](/docs/img/guides/auth-textlocal/2.png)

Reviewer: I would remove this. Getting the API key from the dashboard is pretty self-explanatory without it, and screenshots age very quickly.

> ### Setting a Custom Sender Name (Optional)

Reviewer: ```suggestion
### Customize your sender name (Optional)
```

> Textlocal defaults to 'TXTLCL' as the sender name for all messages. However, customizing this to reflect your brand can enhance communication trustworthiness. To change it, navigate in your Textlocal dashboard: **Settings > All Settings > Sender Names**. Here, you can create and apply a sender name that aligns with your brand, adding a personalized touch to your messages.

Reviewer: ```suggestion
Textlocal defaults to `TXTLCL` as the sender name for all messages. You can customize this to better reflect your brand:

1. In your Textlocal dashboard, go to `Settings` > `All Settings` > `Sender Names`.
1. Change your sender name.
```

> ### Integrating with Supabase

Reviewer: ```suggestion
### Configure Supabase
```

To set up Textlocal as your SMS provider in Supabase, follow these steps:

> 1. **Access Supabase Dashboard**: Navigate to your Supabase project dashboard.
> 2. **Authentication Providers**: Go to **Auth > Providers** or directly access the settings [here](https://supabase.com/dashboard/project/_/auth/providers).
> 3. **Select Phone Provider**: From the list of available authentication providers, choose "Phone".
> 4. **Enable Phone Authentication**: Toggle the option to enable the Phone provider.
> 5. **Enter Textlocal Details**: Input your Textlocal API Key and Sender Name into the corresponding Supabase fields.

Reviewer: ```suggestion
1. In your Supabase Dashboard, go to the [Auth Providers section](https://supabase.com/dashboard/project/_/auth/providers).
1. From the list of available authentication providers, select `Phone`.
1. Toggle `Enable Phone Provider`.
1. Under `SMS Provider`, select `Textlocal.`
1. Enter your Textlocal API Key and Sender Name.
1. Customize your SMS Message (optional).
```

#### Customize Your SMS Template

> Customize the OTP SMS template for branding or to include specific instructions. Use the `.Code` variable to insert the OTP.

Reviewer: Replace with above
